src/combinators/map_data.rs

```diff
@@ -78,3 +79,12 @@ where
         self.inner.is_end_stream()
     }
 }
+
+impl<B, F> fmt::Debug for MapData<B, F>
+where
+    B: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.inner.fmt(f)
```

Comment:
I think we should still keep the structure of `MapData`. By directly delegating `MapData` becomes basically invisible.

```suggestion
        f.debug_struct("MapData")
                .field("inner", &self.inner)
                .field("f", &format_args!("{}", std::any::type_name::<F>()))
                .finish()
```